packages/wrangler/src/__tests__/deploy.test.ts

```diff
@@ -4416,6 +4416,108 @@ addEventListener('fetch', event => {});`
 			`);
 		});
 
+		it("should error when deploying an assets-only Worker with unsupported configuration for assets-only", async () => {
+			writeWranglerConfig({
+				assets: {
+					directory: "assets",
+				},
+				send_metrics: true,
+				vars: {
+					MY_VAR: "my var",
+				},
+				services: [
+					{
+						binding: "BINDING_NAME",
+						service: "WORKER_NAME",
+					},
+				],
+				kv_namespaces: [
+					{
+						id: "123",
+						binding: "KV_BINDING_NAME",
+					},
+				],
+			});
+			await expect(runWrangler("deploy")).rejects
+				.toThrowErrorMatchingInlineSnapshot(`
+				[Error: Assets-only Workers do not support the following configuration keys:
+
+				⋅ "send_metrics"
+				⋅ "vars"
+				⋅ "kv_namespaces"
+				⋅ "services"
+
+				Please remove these fields from your configuration file, or configure the "main" field if you are trying to deploy a Worker with assets.]
+			`);
+		});
+
+		it("should error when deploying an assets-only Worker via the --assets flag, with unsupported configuration for assets-only", async () => {
+			writeWranglerConfig({
+				send_metrics: true,
+				vars: {
+					MY_VAR: "my var",
+				},
+				services: [
+					{
+						binding: "BINDING_NAME",
+						service: "WORKER_NAME",
+					},
+				],
+				kv_namespaces: [
+					{
+						id: "123",
+						binding: "KV_BINDING_NAME",
+					},
+				],
+			});
+			await expect(runWrangler("deploy --assets assets")).rejects
+				.toThrowErrorMatchingInlineSnapshot(`
+				[Error: Assets-only Workers do not support the following configuration keys:
+
+				⋅ "send_metrics"
+				⋅ "vars"
+				⋅ "kv_namespaces"
+				⋅ "services"
+
+				Please remove these fields from your configuration file, or configure the "main" field if you are trying to deploy a Worker with assets.]
+			`);
+		});
+
+		it.fails(
```

Comment:
related to https://github.com/cloudflare/workers-sdk/pull/8998/files#r2048784806